network/topics/controller.go

```diff
@@ -131,7 +131,7 @@ func (ctrl *topicsCtrl) UpdateScoreParams(logger *zap.Logger) error {
 		}
 	}
 	if len(errs) > 0 {
-		return fmt.Errorf(errs)
+		return fmt.Errorf("%s", errs)
```

Comment:
can we rewrite error handling here to use https://pkg.go.dev/errors#Join ?

Reply:
Changes are welcome (but maybe in different PR).
Right now I see we need to fix linter ASAP

Reply:
> I guess it's just this, right @nkryuchkov ?

No, because `errs` is string. I think it should be something like this but I don't mind doing it in another PR

```go

func (ctrl *topicsCtrl) UpdateScoreParams(logger *zap.Logger) error {
	if ctrl.scoreParamsFactory == nil {
		return fmt.Errorf("scoreParamsFactory is not set")
	}
	var errs error
	topics := ctrl.ps.GetTopics()
	for _, topicName := range topics {
		topic := ctrl.container.Get(topicName)
		if topic == nil {
			errs = errors.Join(errs, fmt.Errorf("topic %s is not ready; ", topicName)
			continue
		}
		p := ctrl.scoreParamsFactory(topicName)
		if p == nil {
			errs = errors.Join(errs, fmt.Errorf("score params for topic %s is nil; ", topicName)
			continue
		}
		if err := topic.SetScoreParams(p); err != nil {
			errs = errors.Join(errs, fmt.Errorf("could not set score params for topic %s: %d; ", topicName, err)
			continue
		}
	}

	return errs
}
```

Reply:
let's do it here it should be quite straightforward @nkryuchkov @AKorpusenko

Reply:
@moshe-blox, @nkryuchkov updated this